pkg/agent/controller/networkpolicy/fqdn.go

```diff
@@ -76,11 +76,16 @@ func (fs *fqdnSelectorItem) matches(fqdn string) bool {
 // expirationTime of the records, which is the DNS response
 // receiving time plus lowest applicable TTL.
 type dnsMeta struct {
-	expirationTime time.Time
+	//expirationTime time.Time
 	// Key for responseIPs is the string representation of the IP.
 	// It helps to quickly identify IP address updates when a
 	// new DNS response is received.
-	responseIPs map[string]net.IP
+	responseIPs map[string]ipWithTTL
```

Comment:
would it be simpler if we just keep `map[string]time.Time`?

Reply:
Yes , it can be an improvement but i was wondering that we do have certain scenarios currently , like following below, 

```getIPsForFQDNSelectors()```

```go

		for fqdn := range fqdnsMatched {
			if dnsMeta, ok := f.dnsEntryCache[fqdn]; ok {
				for _, ipWithMetaData := range dnsMeta.responseIPs {
					matchedIPs = append(matchedIPs, ipWithMetaData.ip) <<<<<<
				}
			}
		}
```
where we will have to convert the ip string to ```net.IP``` type and even in that case we will have to make a decision to consider the scenario of conversion  to either ipv4 and ipv6 . So, i thought that ```ipWithTTL``` custom type may come handy in such cases and maybe for future cases too.

Reply:
Actually even the consumers of `getIPsForFQDNSelectors` doesn't really need net.IP. But I agree this would introduce too many changes which is out of the patch's scope. Please keep your current way, we might update all these together in the future.

Reply:
In addition to what Quan suggested, we could switch from net.IP to netip.Addr for a more efficient / compact representation of IP addresses.